cmd/repl/repl_test.go

```diff
@@ -26,12 +27,164 @@ import (
 
 	"github.com/stretchr/testify/assert"
 
-	prompt "github.com/c-bata/go-prompt"
+	"github.com/c-bata/go-prompt"
+	"sqlflow.org/sqlflow/pkg/database"
+	"sqlflow.org/sqlflow/pkg/sql"
+	"sqlflow.org/sqlflow/pkg/sql/testdata"
 )
 
-// TODO(shendiaomo): end to end tests like sqlflowserver/main_test.go
-
 var space = regexp.MustCompile(`\s+`)
+var dbConnStr = "mysql://root:root@tcp(127.0.0.1:3306)/?maxAllowedPacket=0"
+var testDBDriver = os.Getenv("SQLFLOW_TEST_DB")
+var session = makeSessionFromEnv()
+
+func prepareTestData(t *testing.T) error {
+	assertConnectable(dbConnStr)
+	testDB, _ := database.OpenAndConnectDB(dbConnStr)
+	if testDBDriver == "mysql" {
+		_, e := testDB.Exec("CREATE DATABASE IF NOT EXISTS sqlflow_models;")
+		if e != nil {
+			return e
+		}
+		return testdata.Popularize(testDB.DB, testdata.IrisSQL)
+	}
+	t.Skip("Skipping mysql tests")
+	return nil
+}
+
+func getStdout(f func() error) (out string, e error) {
+	oldStdout, oldStderr := os.Stdout, os.Stderr // keep backup of the real stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+	os.Stderr = w
+	e = f() // f prints to stdout
+	outC := make(chan string)
+	go func() { // copy the output in a separate goroutine so printing can't block indefinitely
+		var buf bytes.Buffer
+		io.Copy(&buf, r)
+		outC <- buf.String()
+	}()
+	w.Close()                                   // Cancel redirection
+	os.Stdout, os.Stderr = oldStdout, oldStderr // restoring the real stdout and stderr
+	out = <-outC
+	return
+}
+
+func TestSwitchDatabase(t *testing.T) {
+	a := assert.New(t)
+	prepareTestData(t)
+	session.DbConnStr = dbConnStr
+	results := []interface{}{}
+	for r := range sql.RunSQLProgram("show tables", "", session).ReadAll() {
```

Comment:
`sql.RunSQLProgram` is testing the `sql` package, since below case is already testing `runStmt` maybe we can just remove this test case.

Reply:
Done.